raiden/tests/integration/long_running/test_settlement.py

```diff
@@ -830,3 +835,213 @@ def run_test_automatic_dispute(raiden_network, deposit, token_addresses):
+
+    hold_event_handler.hold_secretrequest_for(
+        secrethash=alice_transfer_secrethash,
+    )
+    hold_event_handler.hold_secretrequest_for(
+        secrethash=bob_transfer_secrethash,
+    )
+
+    alice_app.raiden.start_mediated_transfer_with_secret(
+        token_network_identifier=token_network_identifier,
+        amount=alice_to_bob_amount,
+        fee=0,
+        target=bob_app.raiden.address,
+        identifier=identifier,
+        secret=alice_transfer_secret,
+    )
+
+    bob_app.raiden.start_mediated_transfer_with_secret(
+        token_network_identifier=token_network_identifier,
+        amount=alice_to_bob_amount,
+        fee=0,
+        target=alice_app.raiden.address,
+        identifier=identifier + 1,
+        secret=bob_transfer_secret,
+    )
+
+    wait_for_state_change(
+        raiden=alice_app.raiden,
+        item_type=ActionInitTarget,
+        attributes={},
+        retry_timeout=retry_timeout,
+    )
+
+    wait_for_state_change(
+        raiden=bob_app.raiden,
+        item_type=ActionInitTarget,
+        attributes={},
+        retry_timeout=retry_timeout,
+    )
+
+    alice_bob_channel_state = get_channelstate(alice_app, bob_app, token_network_identifier)
+    alice_lock = channel.get_lock(alice_bob_channel_state.our_state, alice_transfer_secrethash)
+    bob_lock = channel.get_lock(alice_bob_channel_state.partner_state, bob_transfer_secrethash)
+
+    # This is the current state of protocol:
+    #
+    #    A -> B LockedTransfer
+    #    B -> A SecretRequest
+    #    - protocol didn't continue
+    assert_synced_channel_state(
+        token_network_identifier=token_network_identifier,
+        app0=alice_app,
+        balance0=deposit,
+        pending_locks0=[alice_lock],
+        app1=bob_app,
+        balance1=deposit,
+        pending_locks1=[bob_lock],
+    )
+
+    # A ChannelClose event will be generated, this will be polled by both apps
+    # and each must start a task for calling settle
+    RaidenAPI(bob_app.raiden).channel_close(
+        registry_address=registry_address,
+        token_address=token_address,
+        partner_address=alice_app.raiden.address,
+    )
+
+    secret_registry_proxy = alice_app.raiden.chain.secret_registry(
+        secret_registry_address,
+    )
+    secret_registry_proxy.register_secret(
+        secret=alice_transfer_secret,
+        given_block_identifier='latest',
+    )
+    secret_registry_proxy = bob_app.raiden.chain.secret_registry(
+        secret_registry_address,
+    )
+    secret_registry_proxy.register_secret(
+        secret=bob_transfer_secret,
+        given_block_identifier='latest',
+    )
+
+    waiting.wait_for_settle(
+        raiden=alice_app.raiden,
+        payment_network_id=registry_address,
+        token_address=token_address,
+        channel_ids=[alice_bob_channel_state.identifier],
+        retry_timeout=alice_app.raiden.alarm.sleep_time,
+    )
+
+    alice_app.stop()
```

Comment:
yeah, here is the race condition, please stop `alice_app` before the channel settles, and just let `bob_app` settle the channel. 

This is necessary because there is no synchronization from the time `wait_for_settle` and the processing of the events for the corresponding, IOW the unlock transaction can be already sent before this line, making the test flaky.